renku/core/management/migrations/m_0009__new_metadata_storage.py

```diff
@@ -226,6 +232,10 @@ def extract_run_uuid(run_id: str) -> str:
 
     plan_id = Plan.generate_id(uuid=uuid)
 
+    def get_mime_type(path_str: str) -> List[str]:
+        path = Path(path_str)
+        return [DIRECTORY_MIME_TYPE] if path.exists() and path.is_dir() else ["application/octet-stream"]
```

Comment:
since e.g. `output.produces` is either `v9.Entity` or `v9.Collection`, we could fall back to using that if not `path.exists()`.

Or use it directly without `path.is_dir()` (in case it used to be a directory but is a file now)